integration/capabilities/capabilities_test.go

```diff
@@ -0,0 +1,121 @@
+package capabilities
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/docker/testutil/fakecontext"
+
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/poll"
+)
+
+func TestWithFileCapabilities(t *testing.T) {
+
+	withFileCapability := `
+		FROM debian:bullseye-slim
+		RUN apt-get update && apt-get install -y libcap2-bin --no-install-recommends
+		RUN setcap CAP_DAC_OVERRIDE=+eip /bin/cat
+		RUN echo "hello" > /txt
+		RUN chown 0:0 /txt
+		RUN chmod 700 /txt
+		RUN useradd -u 1500 test
+	`
+
+	testCases := []struct {
+		doc        string
+		dockerFile string
+		opts       []func(*container.TestContainerConfig)
+		stdOut     string
+		stdErr     string
+	}{
+		{
+			doc:        "Non-root user with file capability and with capability requested",
+			dockerFile: withFileCapability,
+			opts: []func(*container.TestContainerConfig){
+				container.WithUser("test"),
+				container.WithCapability("CAP_DAC_OVERRIDE"),
+				// TODO where is the nonewprivilages specified
```

Comment:
It is specified by appending the string `"no-new-privileges=true"` to `(*container.TestContainerConfig).HostConfig.SecurityOpt`. Looks like you'll have to implement a new option for it.

```go
func WithSecurityOpt(opt string) func(*TestContainerConfig) {
	return func(c *TestContainerConfig) {
		c.HostConfig.SecurityOpts = append(c.HostConfig.SecurityOpts, opt)
	}
}
```
```suggestion
				container.WithSecurityOpt("no-new-privileges=true")
```